internal/p2p/transport_test.go

```diff
@@ -46,23 +46,18 @@ func TestTransport_AcceptClose(t *testing.T) {
 
 	withTransports(ctx, t, func(ctx context.Context, t *testing.T, makeTransport transportFactory) {
 		a := makeTransport(t)
-		opctx, opcancel := context.WithCancel(ctx)
-
-		// In-progress Accept should error on concurrent close.
-		errCh := make(chan error, 1)
-		go func() {
-			time.Sleep(200 * time.Millisecond)
-			opcancel()
-			errCh <- a.Close()
-		}()
+		opctx, opcancel := context.WithTimeout(ctx, 200*time.Millisecond)
+		defer opcancel()
 
 		_, err := a.Accept(opctx)
 		require.Error(t, err)
 		require.Equal(t, io.EOF, err)
-		require.NoError(t, <-errCh)
+
+		<-opctx.Done()
+		require.Error(t, a.Close())
 
 		// Closed transport should return error immediately.
-		_, err = a.Accept(opctx)
+		_, err = a.Accept(ctx)
```

Comment:
It's probably worth noting the subtlety here, that we use `ctx` specifically because we need an _unterminated_ context.

Reply:
I added some comment